girder/api/rest.py

```diff
@@ -170,7 +170,8 @@ def retVal(user, token):
         return retVal(None, token)
     else:
         try:
-            ensureTokenScopes(token, TokenScope.USER_AUTH)
+            ensureTokenScopes(token, getattr(
+                cherrypy.request, 'requiredScopes', TokenScope.USER_AUTH))
```

Comment:
I understand, but I'd argue that if `cherrypy.request.requiredScopes` is not set, something has gone horribly wrong and we *should* fail. Errors are not a bad thing, if you make a system that doesn't throw them, you end up with the mess that is HTML parsing. I worry that this sort of thing allows other bugs to fester, because they don't just throw an error quickly and loudly, they do some unintended thing silently. IMHO, defence in depth is more about gracefully handling errors from detached subsystems that we are willing to tolerate failure from.

I'd say that we decide "`cherrypy.request.requiredScopes` must now always be set by the time it reaches Girder's handlers", and then enforce that rule. Unless that's not a sensible rule, in which case nevermind everything. :wink: